### ISSUE TITLE
PUT /v1/messages/{destination}

### ISSUE THREAD

**Issue by Laursen79** [Diesel-Jeans/signal#35]
When a message is received on this endpoint, we need to check if the receiving user is present. this can be checked using the client presence logic implemented by #61.

If the client is not present the real server handles it by invoking a notifications manager, but this is out of scope right now. How we handle this scenario is uncertain for the moment.

If the client is present `MessageHandler` should be given the messages which talks with the message cache about caching the message

- [ ] insert messages into storage

This is blocked by #61 and possible implementations of `MessageHandler` and `MessageCache`

**Comment by UnluckyBird** [comment]
Finished in #95
